projects/igniteui-angular/src/lib/icon/icon.references.ts

```diff
@@ -0,0 +1,598 @@
+/** READ BEFORE YOU MODIFY THIS FILE!
+ *
+ * Before you add/modify an icon reference, please think about the semantics of the icon you are adding/modifying.
+ *
+ * Icon aliases have sematic meaning depending on the context in which they are used.
+ * For instance, if your component handles toggling between expanded and collapsed states,
+ * you may want to use the already existing `expand` and `collapse` aliases that point to
+ * the `expand_more` and `expand_less` icons in the material font set.
+ *
+ * It may so happen, however, that the design of your component requires you to use the `chevron_right` for the
+ * expand icon and the `expand_more` for the collapse icon. In this case the `tree_expand` and `tree_collapse` aliases
+ * would be appropriate.
+ * This distinction is important when choosing which icon to use for your component as it will have an impact
+ * when a user decides to rewire the `expand`/`collapse` icons to some other icons.
+ *
+ * Likewise, modifying existing references should be handled with caution as many component in the framework already
+ * share icons that have equivalent semantic meaning. For example, the `Paginator`, `Grid Filtering Row`,
+ * and `Tabs` components in Ignite UI for Angular all use the `prev` and `next` icons for navigating between pages
+ * or lists of items. Changing the underlying target for those icons should be done in a way that suits all components.
+ *
+ * Keep in mind that icon aliases and their underlying names are shared between Ignite UI component frameworks
+ * and changing an alias name here should be reflected in the other frameworks as well.
+ *
+ * To get acquainted with which component uses what icon, please make sure to read the
+ * [docs](https://infragistics.com/products/ignite-ui-angular/Angular/components/icon-service#internal-usage).
+ */
+import { IconMeta } from "./types";
+import type { IconReference, IconThemeKey, MetaReference } from './types';
+
+type Icon = { [key in IconThemeKey]?: IconMeta };
+
+export const iconReferences: MetaReference[] = [];
+
+const makeIconRefs = (icons: Icon) => {
+  return new Map(
+    Object.entries(icons).map((icon) => {
+      return icon as [theme: IconThemeKey, IconReference];
+    })
+  );
+};
+
+const addIcon = (name: string, target: Icon) => {
+    const icon = {
+        alias: {
+            name,
+            family: 'default'
+        },
+        target: makeIconRefs(target)
+    };
+
+    iconReferences.push(icon as MetaReference);
+}
+
+addIcon('more_vert', {
+    default: {
+        name: 'more_vert',
+        family: 'material',
+    }
+});
```

Comment:
Unfortunately this doesn't tree-shake very well 😶
I guess understandable - top level function calls don't are assumed to be a possible side effect and the tree-shake bails.
And those can't really be marked at pure calls either (will permanently be dropped) since they really have a side effect we need when the references are in use + it's mildly infuriating to have to mark each call.

Two alternatives I can think of
- either leave this as a straight up object and give up and give up on the Map convenience (the "default" family assign can live in the service)
- Move the loading of icons in a separate function so it's call can be marked as pure, something like:
![image](https://github.com/user-attachments/assets/f30d3828-9492-4147-96b9-79d76901beb5)
  That's the one setup that I've landed on that does the job. And no, spreading the calls in the var array directly doesn't imply the calls are pure and doesn't allow shaking.
  

PS: Working off of https://github.com/javascript-compiler-hints/compiler-notations-spec/blob/main/pure-notation-spec.md that seems to have more support than the side-effect flag.